app/js/lib/view_handler/element_visibility_switcher.js

```diff
@@ -0,0 +1,118 @@
+'use strict';
+
+var objectAssign = require( 'object-assign' ),
+
+	/**
+	 * View Handler for showing and hiding elements if the update value matches a regular expression
+	 * @class
+	 */
+	VisibilitySwitcher = {
+		showOnValueMatchRegex: null,
+		animator: null,
+
+		isHidden: null, // Null to always trigger show/hide on first call to update
+
+		update: function ( value ) {
+			if ( this.showOnValueMatchRegex.test( value ) ) {
+				if ( this.isHidden === false ) {
+					return;
+				}
+				this.animator.showElement();
+				this.isHidden = false;
+			} else {
+				if ( this.isHidden === true ) {
+					return;
+				}
+				this.animator.hideElement();
+				this.isHidden = true;
+			}
+		}
+	},
+
+	ElementSlideAnimator = {
+		el: null,
+
+		// internal fields
+		slideSpeed: 600,
+
+		showElement: function () {
+			this.el
+				.slideDown( this.slideSpeed )
+				.animate(
+					{ opacity: 1 },
+					{ queue: false, duration: this.slideSpeed }
+				);
+		},
+		hideElement: function () {
+			this.el
+				.slideUp( this.slideSpeed )
+				.animate(
+					{ opacity: 0 },
+					{ queue: false, duration: this.slideSpeed }
+				);
+		}
+	},
+
+	ElementHideAnimator = {
```

Comment:
Like that you can indicate that the animator field of VisibilitySwitcher needs to be of that type. I'd go for `ElementAnimator` (without `Base` prefix). Do you run into single inheritance problems if you do this in JS?